### PR TITLE
Move AnvilCompilationException into the compiler-api module.

### PR DESCRIPTION
**Pull request by vRallev**

### CODE REVIEW

File: compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.jetbrains.kotlin.ir.declarations.IrClass
+import org.jetbrains.kotlin.ir.symbols.IrSymbol
+import org.jetbrains.kotlin.ir.util.render
+import org.jetbrains.kotlin.js.resolve.diagnostics.findPsi
+import org.jetbrains.kotlin.resolve.source.KotlinSourceElement
+import org.jetbrains.kotlin.resolve.source.getPsi
+import org.jetbrains.kotlin.util.getExceptionMessage
+
+/**
+ * `AnvilCompilationException` is thrown when Anvil specific can't be processed or an error
```
Comment (0xMatthewGroves) [inline compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt:18]: This reads strangely. What does "when Anvil specific can't be processed" mean?

File: compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+ * occurs while generating code.
+ */
+public class AnvilCompilationException(
+  message: String,
+  cause: Throwable? = null,
+  element: PsiElement? = null
+) : CompilationException(message, cause, element) {
+  public companion object {
```
Comment (0xMatthewGroves) [inline compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt:26]: Aren't the public visibility operators redundant?

File: compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    ): AnvilCompilationException {
+      return AnvilCompilationException(
+        message = message,
+        cause = cause,
+        element = classDescriptor.identifier
+      )
+    }
+
+    public operator fun invoke(
+      message: String,
```
Comment (0xMatthewGroves) [inline compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt:52]: The ordering isn't standardized across these invocations. Could you align these? 

message
cause
object

**Review by 0xMatthewGroves — APPROVED** [review]
A couple small nits but otherwise LGTM.

File: compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt (hunk 1 above)
Comment (vRallev) [reply compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt:18]: I forgot to type the word "code", "when Anvil specific code can't be processed".

File: compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt (hunk 2 above)
Comment (vRallev) [reply compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt:26]: No, they're required with the explicit API mode enabled.

File: compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt (hunk 3 above)
Comment (vRallev) [reply compiler-api/src/main/java/com/squareup/anvil/compiler/api/AnvilCompilationException.kt:52]: Oh, good catch!

**Pull request merged**